volatility3/framework/plugins/windows/psxview.py

```diff
@@ -0,0 +1,279 @@
+
+    def check_thrdscan(self):
+        ret = []
+
+        for ethread in thrdscan.ThrdScan.scan_threads(
+            self.context, module_name="kernel"
+        ):
+            process = None
+            try:
+                process = ethread.owning_process()
+                if not process.is_valid():
+                    continue
+
+                ret.append(process)
+            except AttributeError:
+                vollog.log(
+                    constants.LOGLEVEL_VVV,
+                    "Unable to find the owning process of ethread",
+                )
+
+        return self.proc_list_to_dict(ret)
+
+    def check_csrss_handles(self, tasks, layer_name, symbol_table):
+        ret = []
+
+        for p in tasks:
+            name = self.proc_name_to_string(p)
+            if name == "csrss.exe":
+                try:
+                    if p.has_member("ObjectTable"):
+                        handles_plugin = handles.Handles(
+                            context=self.context, config_path=self.config_path
+                        )
+                        hndls = list(handles_plugin.handles(p.ObjectTable))
+                        for h in hndls:
+                            if (
+                                h.get_object_type(
+                                    handles_plugin.get_type_map(
+                                        self.context, layer_name, symbol_table
+                                    )
+                                )
+                                == "Process"
+                            ):
+                                ret.append(h.Body.cast("_EPROCESS"))
+
+                except exceptions.InvalidAddressException:
+                    vollog.log(
+                        constants.LOGLEVEL_VVV, "Cannot access eprocess object table"
+                    )
+
+        ret = self.filter_garbage_procs(ret)
+        return self.proc_list_to_dict(ret)
+
+    def check_session(self, pslist_procs):
+        procs = [p for p in pslist_procs if p.get_session_id() != None]
+
+        return self.proc_list_to_dict(procs)
+
+    def _generator(self):
+        kernel = self.context.modules[self.config["kernel"]]
+
+        layer_name = kernel.layer_name
+        symbol_table = kernel.symbol_table_name
+
+        kdbg_list_processes = list(
+            pslist.PsList.list_processes(
+                context=self.context, layer_name=layer_name, symbol_table=symbol_table
+            )
+        )
+
+        processes = {}
+
+        processes["pslist"] = self.check_pslist(kdbg_list_processes)
+        processes["psscan"] = self.check_psscan(layer_name, symbol_table)
+        processes["thrdscan"] = self.check_thrdscan()
+        processes["csrss"] = self.check_csrss_handles(
+            kdbg_list_processes, layer_name, symbol_table
+        )
+        processes["sessions"] = self.check_session(kdbg_list_processes)
+
+        seen_offsets = set()
+        for source in processes:
```

Comment:
The original plugin goes through the list by offset and then by source.  I think that makes more sense, see more a little later on...